nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPath.java

```diff
@@ -255,6 +259,78 @@ public static AccessPath getAccessPathForNodeWithMapGet(Node node, @Nullable Typ
     }
   }
 
+  /**
+   * Constructs an access path ending with the class field element in the argument. The receiver is
+   * the method receiver itself.
+   *
+   * @param element the receiver element.
+   * @return access path representing the class field
+   */
+  public static AccessPath fromElement(Element element) {
+    assert element.getKind().isField()
+        : "element must be of type: FIELD but received: " + element.getKind();
+    Root root = new Root();
+    return new AccessPath(root, Collections.singletonList(new AccessPathElement(element)));
+  }
+
+  /**
+   * Extends an access path with a class field element.
+   *
+   * @param receiverNode the receiver node
+   * @param field element of the class field
+   * @return the extended access path
+   */
+  public static @Nullable AccessPath extendReceiverNodeAccessPathWithField(
+      @Nonnull Node receiverNode, Element field) {
+    if (receiverNode.getTree() == null) {
+      // "this" is the receiver.
+      return fromElement(field);
+    }
+    return extendReceiverTreeAccessPathWithField(receiverNode.getTree(), field);
+  }
+
+  /**
+   * Extends an access path with a class field element.
+   *
+   * @param receiverTree the receiver tree
+   * @param field element of the class field
+   * @return the extended access path
+   */
+  public static @Nullable AccessPath extendReceiverTreeAccessPathWithField(
+      @Nonnull Tree receiverTree, Element field) {
+    if (receiverTree.getKind().equals(Tree.Kind.CONDITIONAL_EXPRESSION)) {
+      // since we can't reason which branch executes, we cannot create an access path for the
+      // expression.
+      return null;
+    }
+    List<AccessPathElement> receivers = getReceiverAccessPathElementChain(receiverTree);
+    receivers.add(new AccessPathElement(field));
+    Root root = new Root(receivers.get(0).getJavaElement());
```

Comment:
What happens here with a tree rooted at `this`, like `this.f.g`?  Will we get the correct `Root`?

Reply:
Yes, I modified the code for ```Root``` class. If we pass ```this``` or ```super``` as the element, the root will still have the following values:
```java
isMethodReceiver = true;
varElement = null;
```

However, I still haven't figure out how to detect reserved keywords like ```this/super``` from an element and still use the ```simpleName``` method to do that.